lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -358,6 +416,22 @@ def set_information(self, trans, id, payload={}, **kwd):
             form_values = trans.model.FormValues(user_info_form, user_info_values)
             trans.sa_session.add(form_values)
             user.values = form_values
+
+        # Update values for extra user preference items
+        extra_user_pref_data = dict()
+        get_extra_pref_keys = self._get_extra_user_preferences( trans )
+        if get_extra_pref_keys is not None:
+            for key in get_extra_pref_keys:
+                key_prefix = key + '|'
+                for item in payload:
+                    if item.startswith( key_prefix ):
+                        # Show error message if the required field is empty
+                        if( payload[item] == "" ):
+                            if( self._check_if_field_required( trans, item ) ):
```

Comment:
We probably should remove `_check_if_field_required` and directly inject its code here, then `get_extra_pref_keys` can be used instead of re-reading the preference file within the loop through `_check_if_field_required`.